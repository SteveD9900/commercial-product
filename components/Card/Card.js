import React, { useState, useEffect } from "react";

import styles from "./Card.module.scss";

export default function Card(props) {
  const [isSale] = useState(props.details.isSale);

  return (
    <div className={styles.Card}>
        {isSale ? (
            <div className={styles.saleStatus}>Sale</div>
        ) : null}
      <div className={styles.content}>
        <div className={styles.imgContainer}>
            <img
                className={styles.displayImage}
                alt="item"
                src={props.details.productImage}
            />
        </div>
        <span className={styles.displayName}>{props.details.productName}</span> &nbsp;
        <span className={styles.displayPrice}>{props.details.price}</span>
      </div>
    </div>
  );
}
