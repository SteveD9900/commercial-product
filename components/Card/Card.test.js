import Card from './Card';
import { shallow } from 'enzyme'

let wrapper;
let detailInfo = {
    "index": 0,
    "isSale": false,
    "price": "$49.99",
    "productImage": "Product_1.jpeg",
    "productName": "Pure Blonde Crate",
    "type": "Beer"
};

beforeEach(() => {
    wrapper = shallow(<Card details={detailInfo}/>);
});

describe('<Card /> rendering', () => {
    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });
});