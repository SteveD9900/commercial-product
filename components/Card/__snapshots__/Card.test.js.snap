// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> rendering renders correctly 1`] = `ShallowWrapper {}`;
